src/booster.jl

```diff
@@ -99,7 +103,13 @@ function Booster(cache::AbstractVector{<:DMatrix};
     elseif !isempty(model_file)
         load!(b, model_file)
     end
-    setparams!(b; kw...)
+    # automatically use gpu_hist if CuArrays used and we didn't pass an explicit argument
+    tm = if isnothing(tree_method)
+        (!isempty(cache) && all(isgpu, cache)) ? (tree_method="gpu_hist",) : (;)
```

Comment:
```suggestion
    no_explicit_argument = isnothing(tree_method)
    tm = if no_explicit_argument
        cache_is_usable = !isempty(cache) && all(isgpu, cache))
        cache_is_usable ? (tree_method="gpu_hist",) : (;)
```
Some software companies have a rule to avoid comments where possible because comments can become outdated. I suggest to avoid the comment here since it's reasonably easy to do so

Reply:
Sorry, I personally just do not like this sort of thing at all.  To me that's replacing a simple comment with horrendously verbose code that's a lot more unpleasant to read.  I'd have a much harder time arguing about it if it were somebody else's PR, but since it's my own I'm simply going to opt out 😄